backend/prompt_studio/prompt_studio_core/prompt_studio_helper.py

```diff
@@ -750,9 +778,43 @@ def dynamic_indexer(
             profile_manager.chunk_size = 0
 
         try:
+
             usage_kwargs = {"run_id": run_id}
             util = PromptIdeBaseTool(log_level=LogLevel.INFO, org_id=org_id)
             tool_index = Index(tool=util)
+            doc_id_key = tool_index.generate_file_id(
+                tool_id=tool_id,
+                vector_db=vector_db,
+                embedding=embedding_model,
+                x2text=x2text_adapter,
+                chunk_size=str(profile_manager.chunk_size),
+                chunk_overlap=str(profile_manager.chunk_overlap),
+                file_path=file_path,
+                file_hash=None,
+            )
+            indexed_doc_id = get_indexed_document_id(doc_id_key)
+            if indexed_doc_id:
+                return indexed_doc_id
+
+            # Polling if document is already being indexed
+            if is_document_indexing(doc_id_key):
+                max_wait_time = 1800  # 30 minutes
+                wait_time = 0
+                polling_interval = 5  # Poll every 5 seconds
+                while is_document_indexing(doc_id_key):
```

Comment:
Is this polling mechanism mandatory? . Will this hold the thread/resources ?

Reply:
Yes, Without this pooling mechanism we end up making multiple indexing for the same document.